src/app/shared/axiosHelper.js

```diff
@@ -1,5 +1,56 @@
+const axios = require("axios");
+const https = require("https");
+const { API_BASE_URL } = require("../../lib/config");
+const { getMiddlewareErrorHandlerMessage } = require("./loggerHelper");
+
+const createAxiosInstance = (onRejected) => {
+  const instance = axios.create({
+    baseURL: API_BASE_URL,
+    httpsAgent: new https.Agent({ keepAlive: true }),
+  });
+
+  instance.interceptors.response.use(null, onRejected);
+
+  return instance;
+};
+
+const axiosErrorHandler = (error) => {
+  const logger = error.config.logger;
+
+  if (axios.isAxiosError(error)) {
+    if (error.response) {
+      const message = getMiddlewareErrorHandlerMessage(
+        error.response,
+        "Error response received in coreBackService",
+      );
+
+      logger.error({ message, level: "ERROR" });
+    } else if (error.request) {
+      const message = {
+        request: error.request,
+        description: "Error occured making request in coreBackService",
```

Comment:
I think this is losing the actual error message?

I think we can be specific about what you want to log here and above, since we know exactly what shape the error has. Perhaps something like:
```
if (axios.isAxiosError(error)) {
  const { request, response } = error;
  if (response) {
    const message = {
      description: "Error response received in coreBackService",
      errorMessage: error.message,
      endpoint: `${request.method} ${request.url}`,
      status: response.status,
      response: response.data,
    };

    logger.error({ message, level: "ERROR" });
  } else if (request) {
    const message = {
      description: "Error occured making request in coreBackService",
      errorMessage: error.message,
      endpoint: `${request.method} ${request.url}`,
    };

    logger.error({ message, level: "ERROR" });
  } else {
    const message = {
      description: "Error occurred setting up request in coreBackService",
      errorMessage: error.message,
    };

    logger.error({ message, level: "ERROR" });
  }
```
Is there anything else in the axios object that you think will be useful?